tests/unit/lms/security_test.py

```diff
@@ -327,63 +365,60 @@ def test_it_returns_LTIUsers_from_authorization_form_fields(
         self,
-        lti11_auth_schema,
         bearer_token_schema,
         OAuthCallbackSchema,
-        canvas_oauth_callback_schema,
+        oauth_callback_schema,
         pyramid_request,
     ):
-        lti11_auth_schema.lti_user.side_effect = ValidationError(["TEST_ERROR_MESSAGE"])
         bearer_token_schema.lti_user.side_effect = ValidationError(
             ["TEST_ERROR_MESSAGE"]
         )
 
         lti_user = _get_lti_user(pyramid_request)
 
         OAuthCallbackSchema.assert_called_once_with(pyramid_request)
-        canvas_oauth_callback_schema.lti_user.assert_called_once_with()
-        assert lti_user == canvas_oauth_callback_schema.lti_user.return_value
-
-    def test_it_returns_LTIUser_from_openid_auth_schema(
-        self, LTI13AuthSchema, lti13_auth_schema, pyramid_request
-    ):
-        pyramid_request.params["id_token"] = "JWT"
-
-        lti_user = _get_lti_user(pyramid_request)
-
-        LTI13AuthSchema.assert_called_once_with(pyramid_request)
-        lti13_auth_schema.lti_user.assert_called_once()
-        assert lti_user == lti13_auth_schema.lti_user.return_value
+        oauth_callback_schema.lti_user.assert_called_once_with()
+        assert lti_user == oauth_callback_schema.lti_user.return_value
 
     def test_it_returns_None_if_all_schemas_fail(
-        self,
-        lti11_auth_schema,
-        bearer_token_schema,
-        canvas_oauth_callback_schema,
-        pyramid_request,
+        self, bearer_token_schema, oauth_callback_schema, pyramid_request
     ):
-        lti11_auth_schema.lti_user.side_effect = ValidationError(["TEST_ERROR_MESSAGE"])
         bearer_token_schema.lti_user.side_effect = ValidationError(
             ["TEST_ERROR_MESSAGE"]
         )
-        canvas_oauth_callback_schema.lti_user.side_effect = ValidationError(
+        oauth_callback_schema.lti_user.side_effect = ValidationError(
             ["TEST_ERROR_MESSAGE"]
         )
 
         assert _get_lti_user(pyramid_request) is None
 
-    def test_LTI11AuthSchema_overrides_BearerTokenSchema(
-        self, lti11_auth_schema, pyramid_request
+    def test_it_stores_the_user(
+        self, pyramid_request, user_service, bearer_token_schema
     ):
-        assert _get_lti_user(pyramid_request) == lti11_auth_schema.lti_user.return_value
-
-    def test_it_stores_the_user(self, pyramid_request, user_service, lti11_auth_schema):
         _get_lti_user(pyramid_request)
 
         user_service.upsert_user.assert_called_once_with(
-            lti11_auth_schema.lti_user.return_value
+            bearer_token_schema.lti_user.return_value
         )
 
+    @pytest.mark.usefixtures("pyramid_request_with_identity")
+    def test_it_picks_lit_user_from_identity(self, pyramid_request):
+        lti_user = _get_lti_user(pyramid_request, from_identity=True)
+
+        assert lti_user == sentinel.lti_user
+
+    @pytest.fixture
+    def pyramid_request_with_identity(self):
+
+        with patch.object(
```

Comment:
This is a bit too much but haven't find a better way

Reply:
TODO: https://docs.pylonsproject.org/projects/pyramid/en/latest/api/config.html#pyramid.config.Configurator.testing_securitypolicy

Reply:
Uses testing_securitypolicy, much cleaner.